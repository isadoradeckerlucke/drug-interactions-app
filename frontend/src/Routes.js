import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import Interactions from "./drugs/Interactions";
import HomePage from "./HomePage";
import Login from "./users/Login";
import Signup from "./users/Signup";
import UserPage from "./users/UserPage";

function Routes() {
  return (
    <Switch>
      <Route exact path="/">
        <HomePage />
      </Route>
      <Route exact path="/interactions">
        <Interactions />
      </Route>
      <Route exact path="/login">
        <Login />
      </Route>
      <Route exact path="/signup">
        <Signup />
      </Route>
      <Route exact path="/users/:username">
        <UserPage />
      </Route>
      <Redirect to="/" />
    </Switch>
  );
}

export default Routes;
