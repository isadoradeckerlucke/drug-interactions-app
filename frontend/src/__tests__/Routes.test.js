import React from "react";
import { render } from "@testing-library/react";
import Routes from "../Routes";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import configureStore from "redux-mock-store";

describe("routes work as expected", function () {
  const currentUserTest = {
    isLoggedIn: true,
    currentUser: {
      username: "testuser",
      email: "testuser@gmail.com",
      savedDrugs: [],
    },
  };

  const initialState = {
    searchTerm: { searchTerm: "adderall" },
    currentUser: currentUserTest,
  };
  const mockStore = configureStore();
  let store;

  it("renders without crashing", function () {
    store = mockStore(initialState);
    render(
      <MemoryRouter>
        <Provider store={store}>
          <Routes />
        </Provider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    store = mockStore(initialState);
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Routes />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
