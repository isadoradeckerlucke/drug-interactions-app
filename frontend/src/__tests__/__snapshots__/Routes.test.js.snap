// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes work as expected matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Welcome to the Drug Interaction Hub, testuser
    </h1>
    <p>
      you haven't saved any medications yet!
    </p>
  </div>
</DocumentFragment>
`;
